### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by aramshiva**
Fixes #4987 partially. It updates the README with correct information on how to setup and some other misc stuff.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <img src="https://cloud-5yru8jas0-hack-club-bot.vercel.app/0logo-512.png" width="126" alt="HCB logo">
   </picture>
   <h1>HCB</h1>
-  <strong>A powerful, safe, and easy-to-use fiscal sponsorship platform for Hack Clubs, hackathons, and nonprofits.</strong>
+  <strong>A powerful, safe, and easy-to-use fiscal sponsorship platform for students to create their own non-profits.</strong>
```
Comment (sampoder) [inline README.md]: I'd like to keep the old text, we could potentially add robotics teams to this list but I think it's nice to keep a direct mention of Hack Clubs and hackathons in here (even if it's just a nice piece of history)

**Review by aramshiva — COMMENTED** [review]
@sampoder I moved it to line 15

**Comment by sampoder** [comment]
Sorry, I don't really see what was wrong with the original

**Comment by aramshiva** [comment]
> Sorry, I don't really see what was wrong with the original

oh my bad I misread

File: README.md [hunk 2]
```diff
@@ -20,64 +20,53 @@ HCB is a platform for managing your nonprofit's finances. Through HCB, you can g
...
+./docker_setup.sh --with-solargraph
 ```
 
-Start the development server with `./docker_start.sh`
-
-Visit [localhost:3000](http://localhost:3000) to see the result.
-
-<!--**What's Solargraph?** [Solargraph](https://solargraph.org/) is a Ruby language server that provides better Intellisense and code completion. It's completely optional to use Solargraph but highly recommended. You may also need to install the [Solargraph extension](https://github.com/castwide/solargraph#using-solargraph) for your IDE.-->
-
-### Production Access
```
Comment (garyhtou) [inline README.md]: Can we keep this "Production Access" section?

Comment (aramshiva) [reply README.md]: I was originally thinking that they would visit the wiki for that but I can update it rn

Comment (garyhtou) [reply README.md]: Ah okay, if this info is already in the Wiki, then it's okay to remove from the readme. We should at least add a link to the wiki page tho.

File: README.md [hunk 3]
```diff
@@ -20,64 +20,69 @@ HCB is a platform for managing your nonprofit's finances. Through HCB, you can g
...
-
-<!--**What's Solargraph?** [Solargraph](https://solargraph.org/) is a Ruby language server that provides better Intellisense and code completion. It's completely optional to use Solargraph but highly recommended. You may also need to install the [Solargraph extension](https://github.com/castwide/solargraph#using-solargraph) for your IDE.-->
+```bash
+./docker_start.sh
+# or with Solargraph
+./docker_start.sh --with-solargraph
+```
 
 ### Production Access
+Then, to start the development server:
```
Comment (garyhtou) [inline README.md]: Did we want this line here?

File: README.md [hunk 4]
```diff
@@ -20,64 +20,69 @@ HCB is a platform for managing your nonprofit's finances. Through HCB, you can g
 - [Table of Contents](#table-of-contents)
 - [Contributing](#contributing)
 - [Quick Start](#quick-start)
-  - [Production Access](#production-access)
+  - [Github Codespaces](#github-codespaces)
```
Comment (garyhtou) [inline README.md]: If we're keeping the prod access section, let's add it back to this table of contents.

File: README.md [hunk 5]
```diff
@@ -20,64 +20,69 @@ HCB is a platform for managing your nonprofit's finances. Through HCB, you can g
...
-```
+To get started, [whip up a codespace](https://docs.github.com/en/codespaces/getting-started/quickstart) and follow the steps in the [Automated Setup with Docker](#automated-setup-with-docker) section.
+
+See the [Codespaces](./Codespaces.md) page for more information on developing in a GitHub Codespaces environment.
 
-Install and run [Docker](https://docs.docker.com/get-docker/).
+### Docker
 
-Run the [docker_dev_setup.sh](./docker_dev_setup.sh) script to install dependencies and set up a local environment with Docker.
+If you are running macOS or Ubuntu, you can clone the repository and run the [docker_setup.sh](./docker_setup.sh) script to automatically set up a development environment with Docker. Append `--with-solargraph` to the command to also setup [Solargraph](https://solargraph.org), a language server for Ruby. You may also need to install the [Solargraph extension](https://github.com/castwide/solargraph#using-solargraph) for your editor.
```
Comment (garyhtou) [inline README.md]: FYI, this script should also work for Windows; although it's recommended that Window users run it (docker) within [WSL](https://learn.microsoft.com/en-us/windows/wsl/install)

**Review by garyhtou — COMMENTED** [review]
just a couple small things!

File: README.md [hunk 6]
```diff
@@ -58,22 +59,7 @@ Then, to start the development server:
...
-    - If you need to edit [`config/credentials/production.yml.enc`](./config/credentials/production.yml.enc), you can now run `bin/rails credentials:edit --environment=production`.
-- Run the [docker_setup.sh](./docker_setup.sh) script to set up a local environment with Docker using a dump of the production database.
-```bash
-./docker_start.sh
-# or with Solargraph
-./docker_start.sh --with-solargraph
-```
-
-Developing with production keys & data has risk and should only be used if you know what you are doing.
+If you have more questions about development check out our [wiki](https://github.com/hackclub/hcb/wiki)
```
Comment (garyhtou) [inline README.md]: Just one last thing. Let's make sure we have a direct link to the production access section in the wiki. I wasn't able to find it from a quick search

```suggestion
If you have more questions about development check out our [wiki](https://github.com/hackclub/hcb/wiki)

### Production Access

Please see this wiki: <INSERT DIRECT LINK TO THE PRODUCTION ACCESS LINK HERE>
```

**Review by garyhtou — APPROVED** [review]
perfect! thank you for this

**Comment by garyhtou** [comment]
In the future, it would be cool if we documented the docker-less setup in the README

**Pull request merged**